### PR TITLE
Update doc for CA1305 to mention excluded types

### LINKED ISSUE #4986: CA1305 -`ToString` call ignored for `bool`, `char` and `guid`.
Hi there,

With the release of fxcop 3.X the analyzer will no longer report on `ToString` calls for `char`, `bool` and `Guid`. I am not sure whether or not you want to update the doc to mention this part.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 5f418cad-87d9-537a-497c-991d209cb178
* Version Independent ID: 10bbdf2e-c425-4f8d-bfec-a5b821a41698
* Content: [CA1305: Specify IFormatProvider - Visual Studio](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1305?view=vs-2019)
* Content Source: [docs/code-quality/ca1305.md](https://github.com/MicrosoftDocs/visualstudio-docs/blob/master/docs/code-quality/ca1305.md)
* Product: **visual-studio-windows**
* Technology: **vs-ide-code-analysis**
* GitHub Login: @Mikejo5000
* Microsoft Alias: **mikejo**

### PR DESCRIPTION
**Pull request by Evangelink**
Fix #4986

This new feature is not yet publicly released. Please sync-up with @mavasani for when to merge this PR.

### CODE REVIEW

**Comment by PRMerger15** [comment]
@Evangelink : Thanks for your contribution! The author(s) have been notified to review your proposed change.

File: docs/code-quality/ca1305.md [hunk 1]
```diff
@@ -35,6 +35,9 @@ This rule ignores calls to .NET methods that are documented as ignoring the <xre
 - <xref:System.Activator.CreateInstance%2A?displayProperty=nameWithType>
 - <xref:System.Resources.ResourceManager.GetObject%2A?displayProperty=nameWithType>
 - <xref:System.Resources.ResourceManager.GetString%2A?displayProperty=nameWithType>
+- <xref:bool.ToString%2A?displayProperty=nameWithType>
+- <xref:char.ToString%2A?displayProperty=nameWithType>
+- <xref:System.Guid.ToString%2A?displayProperty=nameWithType>
```
Comment (mavasani) [inline docs/code-quality/ca1305.md:40]: @Evangelink Can you remind me of the PR on roslyn-analyzers to fix this?

Comment (Evangelink) [reply docs/code-quality/ca1305.md:40]: https://github.com/dotnet/roslyn-analyzers/pull/3379 and https://github.com/dotnet/roslyn-analyzers/pull/2938

Comment (Mikejo5000) [reply docs/code-quality/ca1305.md:40]: @mavasani The changes here look OK to merge?

Comment (mavasani) [reply docs/code-quality/ca1305.md:40]: Yes.

**Comment by Mikejo5000** [comment]
#sign-off Thanks for the doc updates, @Evangelink !

**Pull request merged**